core/components/versionx/model/versionx.class.php

```diff
@@ -399,6 +399,10 @@ public function getVersionDetails($class = 'vxResource',$id = 0, $json = false,
             /* Class specific processing */
             switch ($class) {
                 case 'vxResource':
+                    $resource = $this->modx->getObject('modResource',$v->get('content_id'));
+                    if(!$resource) {
```

Comment:
This check should be removed; just because a resource no longer exists does not mean it should not be available in VersionX. Restoring resources that were deleted is kind of a big deal for a versioning plugin.

Reply:
Yes, you're right. I will check for a better solution later on.